imagebuildah/stage_executor.go

```diff
@@ -2363,7 +2363,17 @@ func (s *StageExecutor) commit(ctx context.Context, createdBy string, emptyLayer
 	return imgID, ref, nil
 }
 
-func (s *StageExecutor) generateBuildOutput(buildOutputOpts define.BuildOutputOption) error {
+func (s *StageExecutor) generateBuildOutput(opts define.BuildOutputOption) error {
+	if opts.Type == define.BuildOutputImage {
+		if opts.Push {
+			err := internalUtil.PushImage(s.executor.store, opts, s.executor.systemContext)
+			if err != nil {
+				return fmt.Errorf("failed to export build output: %w", err)
+			}
+		}
+		return nil
```

Comment:
Is this right it does no output if not opts.Push?

Reply:
iiuc, `type=image` outputs a container image format, only when you specify `--output`, it additionally pushes the image to the registry.